libraries/stdlib/stdlib_defs.mtlx

```diff
@@ -2761,6 +2761,18 @@
     <output name="out" type="vector3" />
   </nodedef>
 
+  <!--
+    Node: <refract>
+    Compute the refraction direction of an input vector against a normal vector.
+    The normal vector is assumed to be normalized.
+  -->
+  <nodedef name="ND_refract_vector3" node="refract" nodegroup="math" doc="Compute the refraction direction of an input vector.">
+    <input name="in" type="vector3" value="1.0,0.0,0.0" doc="Input vector to be reflected."  />
+    <input name="normal" type="vector3" defaultgeomprop="Nworld" doc="Normal vector." />
+    <input name="eta" type="float" value="1.0" doc="Ratio of the indices of refraction."/>
```

Comment:
Is there an input range that applies here? Assume you don't want 0.

Reply:
Agreed, we want the value to be bigger than 0, but do we have a way to express that in the current metadata?  I only really see `uimin` and `uisoftmin`.  What would we consider a safe epsilon above zero? 

We're a bit inconsistent on  ui hints for IOR in other nodes, `dielectric_bsdf` and other nodes in pbrlib don't provide any.  `standard_surface` and `UsdPreviewSurface` say `uimin="0.0" uisoftmin="1.0"  uisoftmax="3.0"`, `gltf_pbr` says ` uimin="1" uisoftmax="3" `

I'm open to guidance on what to set here.

Reply:
I wouldn't worry about setting these ranges in the initial implementation, but going forward I would recommend the `uisoftmin="1.0"  uisoftmax="3.0"` of UsdPreviewSurface, since there are artistic use cases for values outside this range.